projects/cfengine/string_fuzzer.c

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2021 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <stdint.h>
+#include <string.h>
+#include <stdlib.h>
+#include <string_expressions.h>
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size){
+    if(size<4) {
+        return 0;
+    }
+    for (int i=0; i<size; i++) {
+        if(data[i]==0) {
```

Comment:
I wonder if it would be better to change all 0s to some other constant.
A downside would be that testcases would still have 0s making them harder for maintainers to debug (they would have to know that they get changed to some other constant or use this target.

Hopefully it shouldn't matter but it would be interesting (from a fuzzer author point of view) to see how many tests get thrown out by this check. It seems like it would be hard to generate a string that say needed to be 500 chars without having any 0s (1-1/256)**500=0.14

Reply:
Yea, I guess replacing with a constant would be better, or even better, getting the random number generator to go from 1-255.